src/string.cpp

```diff
@@ -368,6 +368,35 @@ bool StrEndsWith(const std::string_view str, const std::string_view suffix)
 	return str.compare(str.size() - suffix_len, suffix_len, suffix, 0, suffix_len) == 0;
 }
 
+/**
+ * Compares two string( view)s, while ignoring the case of the characters.
+ * @param str1 The first string.
+ * @param str2 The second string.
+ * @return Less than zero if str1 < str2, zero if str1 == str2, greater than
+ *         zero if str1 > str2. All ignoring the case of the characters.
+ */
+int StrCompareIgnoreCase(const std::string_view str1, const std::string_view str2)
+{
+	size_t len = std::min(str1.size(), str2.size());
+	int res = strncasecmp(str1.data(), str2.data(), len);
+	if (res != 0) return res;
+
```

Comment:
If `strncasecmp` returns 0, aren't they considered equal?

Reply:
Due to `std::string_view` you're not guaranteed a '\0', so we must limit it manually to the shortest and compare. When they are equal, then the difference in size will determine the result.

Having said that str(n)casecmp is essentially broken for comparison as it compares ASCII characters when we feed it UTF-8. To properly solve that, we have to go the way of strnatcmp, which uses ICU/OSX/Windows specific APIs for sorting accounting for all these things.

But as stop-gap I can implement a pure C++ variant of strcasecmp.